server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -317,15 +335,20 @@ public Consumer getConsumer(String uuid) {
      * @return
      *  A query to fetch the consumers with the specified consumer IDs
      */
-    public CandlepinQuery<Consumer> getConsumers(Collection<String> consumerIds) {
+    public Collection<Consumer> getConsumers(Collection<String> consumerIds) {
+        Set<Consumer> consumers = new HashSet<>();
+        javax.persistence.Query query = this.getEntityManager()
+            .createQuery("SELECT c FROM Consumer c WHERE c.id IN (:cids)");
```

Comment:
Should this be loaded using session.byMultipleIds(this.entityType()).enableSessionCheck(true).multiLoad(<id list>) similarly to what we are doing in the ProductCurator so the partitioning & checking from the cache can be offloaded to hibernate?

Reply:
The difference in cache usage shouldn't be the same, but I made the change so we can offload the partitioning bit to Hibernate and not worry about yet another problem with the IN operator.